### DISCUSSION TITLE
Need help- Trying To enable ReBar on GTX 980

### DISCUSSION THREAD

**Discussion by SlowerBeen** [terminatorul/NvStrapsReBar#28] [General]
I read some Discussions about people being able to enable rebar on old gpus, but i dont get the dettails because they all happent in a big discussion.

perhaps some one could please explain me the only steps that might work so that i can test them ?
i read about changing the bar1 straps for pre pascal cards, i am just hoping that behind that discussion there is a possiblity
of make it work on 900 series GPU.

i would like some one of you that know more to please give me some help, and fix my confusion.

**Comment by terminatorul** [comment]
You should always start by following the instructions on the [NvStrapsReBar](https://github.com/terminatorul/NvStrapsReBar) page. The instructions also reference the [ReBarUEFI](https://github/xCuri0/ReBarUEFI) page, so follow that as well.

Please provide more information like a GPU-Z screenshot, CPU-Z screenshot of your CPU, MB and GPU, and a screenshot from NvStrapsReBar.exe console window.

Also please report if UEFI patches described in ReBarUEFI project apply to your motherboard (chipset), and if you were able to apply them as instructed.

I don't have this generation of hardware, I assume GTX 980 is Maxwell 2, and I don't have user reports for this hardware either. So your mileage may vary, and I already know some older architectures like Keppler 2 (GK208 chips) and Pascal (GTX1000 generation) are not supported by the NVIDIA driver for Windows. Also Keppler 1 requires some special case to be implemented in NvSteapsReBar (in the DXE driver), but that will be visible later, if it applies to you as well.

**Reply by SlowerBeen** [reply]
I have mi bios patched with uefi patches + USB 3 for Z77 and i have 4g decoding native enabled in bios and i have nvstrap rebar module installed in that bios + CSM disabled.
I tried 512mb bar size, my pc actually boots (can also ear sound from speakers) but its only the gpu that stops sending the signal to the monitor

(My computer works no problem even whit padding issue after UEFI patch, that i was not able to fix whit MMtool work around advertised in the main Rebar guide)


![nvstrap](https://github.com/terminatorul/NvStrapsReBar/assets/161142162/e1078f49-34f8-405b-8fa5-f7b5d1766dd0)
![Gpu z](https://github.com/terminatorul/NvStrapsReBar/assets/161142162/43a55907-7842-4455-b917-3019bcab754b)
![cpu z 1](https://github.com/terminatorul/NvStrapsReBar/assets/161142162/de1b2d76-10c7-4755-8360-5e9f2cfe3a8b)
![cpu z 2](https://github.com/terminatorul/NvStrapsReBar/assets/161142162/370a14a6-2bac-47ed-a0dc-01fde0c3b571)

**Reply by xCuri0** [reply]
@SlowerBeen you could set iGPU to always enabled in BIOS and connect your monitor to it.


Maybe it could help figure out what's going on

**Reply by terminatorul** [reply]
I suspect the NVIDIA driver for Windows does not work with BAR size over 256 MiB for Maxwell 2 cards (same as for Kepler 2).

Boot with iGPU like @xCurio0 said, and look into the output from NvStrapsReBar.exe to see the current BAR size. Is it the same as the target BAR size ? You can also look in Device Manger at the memory resource page.

Instead of iGPU you could also remove the NVIDIA driver from Windows Settings and from Device Manger, until Device Manager shows Microsoft Basic Display Adapter instead of NVIDIA. You will have to disable or disconnect from network (both wired and wireless) to prevent Windows Update from re-installing the NVIDIA driver. You also need to build a debug version of NvStrapsReBar.exe to list and configure non-NVIDIA adapters (that is the Basic Display Adapter)....

If you do this, I suspect you will see a PCI BAR size of 512 MiB for example for the NVIDIA card (or the Microsoft Basic Display Adapter). Can you show screenshots to confirm or infirm this ? If you keep the NVIDIA driver, I suspect you will see a question mark over the graphics adapter icon in Device Manager, and no memory resources will be allocated for the adapter at all.

You can also try an older NVIDIA driver (I think 471 was the last 400 driver....), 'though I am not sure this helps anyone, and chances for this to work are still low.

**Reply by SlowerBeen** [reply]
![image](https://github.com/terminatorul/NvStrapsReBar/assets/161142162/00ba9a9f-bc25-4f8b-b267-fca2d9e0526a)

This is the situation booting whit igpu, i will keep following all your instructions till i can say that i dont have a chance for this. 
thank you all for the answer.

**Reply by terminatorul** [reply]
Device Manager screenshot only shows BAR0. Can you scroll down and find BAR1 ?

If you can't see it, you should uninstall NVIDIA driver and fall back to Microsoft Basic Display Adapter

**Reply by SlowerBeen** [reply]
I made this by just disabling windows update and unistalled nvidia driver.
i havent created a specific version of nvstraprebar for debug. this is the result.
![image](https://github.com/terminatorul/NvStrapsReBar/assets/161142162/59eedff7-bd62-4899-a8f8-1f4a16364a88)

**Reply by SlowerBeen** [reply]
I also just tried to install nvidia driver 471, and after the driver completed install and tock control, the black screen just engaged refusing 512mb  bar size i guess. I may have forgot to set pci bar size to activate bar on mobo side, but i think the problem will be the same, i standby for you higher knowledge. 
Also could you explain whats the difference between bar0 bar1 ecc

**Reply by terminatorul** [reply]
This confirms my expectations.

The memory resources in Device Manager and the screenshot with NvStrapsReBar.exe show the NVIDIA card really has a 512 MiB size for BAR1. The NvStrapsReBar driver did what it was supposed to do.

But then the NVIDIA driver for Windows detects the new BAR size and reports an error to Windows.

I am sorry there is nothing else I can do about this. Like I said I have a similar card (GT 710, with a Kepler chip KG208) and I cannot make it work either.

I don't know exactly what the 3 BARs are used when working with the GPU, but I know BAR0 is used configuration and control (sending commands to the GPU), and BAR1 is used as a memory buffer for vertices in the 3D scene, textures and the rendered image (frame buffer).
